testutil/integ.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"log"
+	"os"
+	"strconv"
+	"testing"
+)
+
+// IsIntegration checks env var TEST_INTEGRATION and consider that we're in an
+// integration test if it's set to true.
+func IsIntegration() bool {
+	integVal := os.Getenv("TEST_INTEGRATION")
+	if integVal == "" {
+		return false
+	}
+	isInteg, err := strconv.ParseBool(integVal)
+	if err != nil {
+		log.Printf("WARNING: failed to parse TEST_INTEGRATION; will default to 'false': %v", err)
```

Comment:
Optional: you might consider returning an error here. If the user provides TEST_INTEGRATION=TRUe (with an erroneous lower-case "e") then the test will be silently skipped, and the user could be misled into thinking that their test passed.

This might be overengineering 😄

Reply:
I've debated with myself :) But now you voted...